scipy/stats/_morestats.py

```diff
@@ -4467,40 +4456,23 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
     >>> plt.show()
 
     """
-    samples, sin_samp, cos_samp, nmask = _circfuncs_common(samples, high, low,
-                                                           nan_policy=nan_policy)
-    sin_sum = sin_samp.sum(axis=axis)
-    cos_sum = cos_samp.sum(axis=axis)
+    samples, sin_samp, cos_samp = _circfuncs_common(samples, high, low)
+    sin_sum = sin_samp.sum()
+    cos_sum = cos_samp.sum()
     res = arctan2(sin_sum, cos_sum)
 
-    mask_nan = ~np.isnan(res)
```

Comment:
Do you agree that all this `mask_nan` stuff was unnecessary even in the original code?

Reply:
Yeah, it wouldn't have made a difference to not have it. I guess the author was just afraid that `nan`s might cause `res < 0` comparison to go wrong.